algorithms/palindrome.py

```diff
@@ -36,3 +36,31 @@ def isPalindrome(self, s: str):
             end -= 1
 
         return True
+
+    # https://leetcode.com/problems/valid-palindrome-ii/
+
+    def check_palindrome(self, s, start, end):
+        while start < end:
+            if s[start] != s[end]:
+                return False
+            start += 1
+            end -= 1
+        return True
+
+    def validPalindrome(self, s: str):
+        start, end = 0, len(s) - 1
+        allowed_to_delete = True
+
+        while start < end:
+            ch_start, ch_end = s[start], s[end]
+            if ch_start != ch_end:
+                if allowed_to_delete:
```

Comment:
I think, you can just remove `allowed_to_delete`, you're returning result right after setting value anyway.